src/indexer/index_writer.rs

```diff
@@ -61,7 +60,7 @@ pub struct IndexWriter {
 
     memory_arena_in_bytes_per_thread: usize,
 
-    workers_join_handle: Vec<JoinHandle<crate::Result<()>>>,
+    workers_future_results: Vec<Mutex<FutureResult<()>>>,
```

Comment:
I had to use a Mutex to be able to clone the `IndexWriter`, useful for multiple producers use case.

Reply:
i don't think we allow cloning IndexWriter today. Do we?

Reply:
In the multiple producer example

Reply:
Indexwriter is in an arc mutex